lib/game.rb

```diff
@@ -1,17 +1,71 @@
 require_relative "board.rb"
+require_relative "coordinates.rb"
 
 class Game
+  attr_reader :std_out, :std_in, :coordinates
+
+  def initialize(std_out, std_in)
+    @std_out = std_out
+    @std_in = std_in
+  end
+
   def self.run(*args)
-    Game.new.run(*args)
+    Game.new(*args).run
   end
 
-  def run(output)
-    output.puts empty_board
+  def run
+    start_game
+    retry_until_valid
+    end_game
   end
 
   private
 
+  def start_game
+    std_out.puts game_start_message
+    set_coordinates_from_user_input
+  end
+
+  def game_start_message
+    empty_board + "\n" + start_prompt + "\n" + instructions_prompt
+  end
+
   def empty_board
     Board.new.to_s
   end
+
+  def start_prompt
+    "Where would you like to play?"
+  end
+
+  def instructions_prompt
+    'Enter a row letter and column number. For Example: "A1".'
+  end
+
+  def retry_until_valid
+    until valid_input?
+      reprompt_input
+    end
+  end
+
+  def valid_input?
+    coordinates.valid?
+  end
+
+  def set_coordinates_from_user_input
+    @coordinates = Coordinates.new(std_in.gets)
```

Comment:
When you run the game for real, can you see it get valid input and exit? `gets` includes the ending newline, so a value coming in here looks like `"A2\n"`, which I think won't pass your validation checks. Generally, scripts in Ruby will do `gets.chomp` to swallow the newline.

Reply:
I swear this was working before! but you're right I needed to add chomp!